JSQMessagesDemo/DemoModelData.m

```diff
@@ -135,9 +135,12 @@ - (void)loadFakeMessages
      *  Setting to load extra messages for testing/demo
      */
     if ([NSUserDefaults extraMessagesSetting]) {
-        NSArray *copyOfMessages = [self.messages copy];
-        for (NSUInteger i = 0; i < 4; i++) {
-            [self.messages addObjectsFromArray:copyOfMessages];
+        NSArray *copyOfMessages = [NSKeyedUnarchiver unarchiveObjectWithData:
+            [NSKeyedArchiver archivedDataWithRootObject:self.messages]];
+        [self.messages addObjectsFromArray:copyOfMessages];
+        for (NSUInteger i = 0; i < 3; i++) {
+            [self.messages addObjectsFromArray:[NSKeyedUnarchiver unarchiveObjectWithData:
+                [NSKeyedArchiver archivedDataWithRootObject:copyOfMessages]]];
```

Comment:
```- initWithArray:copyItems:``` doesn't fix the issue - you can try it in the demo app.  The problem with ```- initWithArray:copyItems:``` is that it makes only one-level deep copy. But our media view is deeper and it's not copied.

Take a look at this discussion:
http://stackoverflow.com/questions/647260/deep-copying-an-nsarray